src/Uno.UI/UI/Xaml/ResourceResolver.cs

```diff
@@ -0,0 +1,323 @@
+			{
+				return false;
+			}
+
+			if (parseContext.AssemblyName == "Uno.UI")
+			{
+				return false;
+			}
+
+			if (_registeredDictionariesByAssembly.TryGetValue(parseContext.AssemblyName, out var assemblyDict))
+			{
+				foreach (var kvp in assemblyDict)
+				{
+					var rd = kvp.Value as ResourceDictionary;
+					if (rd.TryGetValue(resourceKey, out value))
+					{
+						return true;
+					}
+				}
+			}
+
+			return false;
+		}
+
+		/// <summary>
+		/// Get a system-level resource with the given key.
+		/// </summary>
+		/// <remarks>
+		/// Use <see cref="ResolveTopLevelResource{T}(object, T)"/> when user-defined Application-level values should be considered (most
+		/// of the time), use <see cref="GetSystemResource{T}(object)"/> if they shouldn't
+		/// </remarks>
+		internal static T GetSystemResource<T>(object key)
+		{
+			if (MasterDictionary.TryGetValue(key, out var value) && value is T t)
+			{
+				return t;
+			}
+
+			return default(T);
+		}
+
+		/// <summary>
+		/// Push a new <see cref="XamlScope"/>, typically because a template is being materialized.
+		/// </summary>
+		/// <param name="scope"></param>
+		internal static void PushNewScope(XamlScope scope) => _scopeStack.Push(scope);
+		/// <summary>
+		/// Push a new Resources source to the current xaml scope.
+		/// </summary>
+		internal static void PushSourceToScope(DependencyObject source) => PushSourceToScope((source as IWeakReferenceProvider).WeakReference);
+		/// <summary>
+		/// Push a new Resources source to the current xaml scope.
+		/// </summary>
+		internal static void PushSourceToScope(ManagedWeakReference source)
+		{
+			var current = _scopeStack.Pop();
+			_scopeStack.Push(current.Push(source));
+		}
+		/// <summary>
+		/// Pop Resources source from current xaml scope.
+		/// </summary>
+		internal static void PopSourceFromScope()
+		{
+			var current = _scopeStack.Pop();
+			_scopeStack.Push(current.Pop());
+		}
+		/// <summary>
+		/// Pop current <see cref="XamlScope"/>, typically because template materialization is complete.
+		/// </summary>
+		internal static void PopScope()
+		{
+			_scopeStack.Pop();
+			if (_scopeStack.Count == 0)
+			{
+				throw new InvalidOperationException("Base scope should never be popped.");
+			}
+		}
+
+		/// <summary>
+		/// If tracing is enabled, writes an event for the initialization of system-level resources (Generic.xaml etc)
+		/// </summary>
+		internal static IDisposable WriteInitiateGlobalStaticResourcesEventActivity() => _trace.WriteEventActivity(TraceProvider.InitGenericXamlStart, TraceProvider.InitGenericXamlStop);
+
+		/// <summary>
+		/// Register a dictionary for a given source, this is used for retrieval when setting the Source property in code-behind or to an
+		/// external resource.
+		/// </summary>
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public static void RegisterResourceDictionaryBySource(string uri, XamlParseContext context, Func<ResourceDictionary> dictionary)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either change this parameter type to 'System.Uri' or provide an overload which takes a 'System.Uri' parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)